Move oid_wraparound to the bottom of the schedule

huansong opened the pull request:
Updated
===========
The real issue see https://github.com/greenplum-db/gpdb/pull/14899#issuecomment-1435299158

Original PR description
===========

The test oid_wraparound consumes all remaining OIDs and wraps it around. Then, we would be reusing the OIDs we used before. That could lead to flakiness to subsequent tests such as the one we saw in `uao_dml_unique_index_update_column` (could happen to `uao_dml_unique_index_update_row` too):

```
 create schema uao_dml_unique_index_update_ao_column;
+DETAIL:  Key (classid, objid, staactionname)=(2615, 27835, CREATE) already exists. 
+ERROR:  duplicate key value violates unique constraint "pg_statlastop_classid_objid_staactionname_index"
```

We are reusing the same OID of the schema we created in a previous test 'foreign_data':

```sql
regression=# select * from pg_stat_last_operation where objid = 27835;
 classid | objid | staactionname | stasysid |        stausename         | stasubtype |            statime
---------+-------+---------------+----------+---------------------------+------------+-------------------------------
    2615 | 27835 | CREATE        |    27376 | regress_foreign_data_user | SCHEMA     | 2023-02-03 03:31:45.110291+00
(1 row)
```

Now moving the oid_wraparound test to the bottom of the greenplum schedule, which is also the last schedule to run.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

ashwinstar commented:
Would like to learn more why there is flakiness, like test relying on some fixed oid? Having wrap around coverage is great to have in pipeline hence would like to see the real need before we let it go.

huansong commented:
@ashwinstar Here's a bit more detailed recap:
1. Test `foreign_data` created a schema `foreign_schema` which was assigned with OID 27835. This also inserted a record into `pg_stat_last_operation` starting with (classid=2615, objid=27835, staactionname=CREATE,...)
2. Test `foreign_data` dropped `foreign_schema`, so 27835 would be available to use in future.
3. Test `oid_wraparound` wrapped OID around, so the next OID will be starting from 16384.
4. A bunch of other tests ran and consumed OIDs.
5. Test `uao_dml_unique_index_update_column` created a schema which happened to be assigned with OID 27835. It attempted to insert another record `pg_stat_last_operation` starting with (classid=2615, objid=27835, staactionname=CREATE,...). That conflicted with the aforementioned one due to the unique constraint on these columns, hence the error.

Not letting `oid_wraparound` go but just moving it to the end of the test schedule so that it won't cause problem like above. I guess in real life the above error is very unlikely to happen so I was not looking to really "fix" the issue by e.g., removing the unique constraint in `pg_stat_last_operation` (I actually wonder why it is needed though?)

huansong commented:
Found w/ @bmdoil : actually, the real issue here is that we were not dropping the meta tracking rows associated with `foreign_schema` when we dropped it. We would do that when dropping other objects like `DropRole`, `DropTableSpace` etc.

In fact, we used to do the right thing in `RemoveSchema`: https://github.com/greenplum-db/gpdb/commit/c83fb8108854#diff-8e4578a03e7baffa8587f2d972ca742e61090e70652e4938c4b5b2a5703497b9R247-R251 . But that got lost in [a merge later](https://github.com/greenplum-db/gpdb/commit/4750e1b65518a5575e0cc75eb6716cc7d5594139) which got rid of `RemoveSchema` and only used `RemoveSchemaById` ever since.

So the real fix here is to do `MetaTrackDropObject` in `RemoveSchemaById`. Added test case too.

huansong commented:
Probably need to backport this to 6X too.
